discord/sink.py

```diff
@@ -0,0 +1,914 @@
+from dataclasses import dataclass
+import logging
+import os
+import struct
+import subprocess
+import threading
+import wave
+from typing import TYPE_CHECKING, Any, BinaryIO, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+from .enums import RTCPMessageType
+from .errors import ClientException
+from .opus import Decoder as OpusDecoder
+from .player import CREATE_NO_WINDOW
+
+if TYPE_CHECKING:
+    from .member import Member
+    from .voice_client import VoiceClient
+
+
+__all__ = (
+    "AudioFrame",
+    "AudioSink",
+    "AudioFileSink",
+    "WaveAudioFileSink",
+    "MP3AudioFileSink",
+    "RTCPPacket",
+    "RTCPSenderReportPacket",
+    "RTCPReceiverReportPacket",
+    "RTCPSourceDescriptionPacket",
+    "RTCPGoodbyePacket",
+    "RTCPApplicationDefinedPacket",
+    "RTCPReceiverReportBlock",
+    "RTCPSourceDescriptionChunk",
+    "RTCPSourceDescriptionItem",
+)
+
+
+_log = logging.getLogger(__name__)
+
+
+@dataclass
+class RTCPReceiverReportBlock:
```

Comment:
Considering how many of these are made, I feel like these would benefit from having slots